metric_learn/base_metric.py

```diff
@@ -242,6 +242,118 @@ def transform(self, X):
     """
 
 
+class BilinearMixin(BaseMetricLearner, metaclass=ABCMeta):
+  r"""Bilinear similarity learning algorithms.
+
+  Algorithm that learns a Bilinear (pseudo) similarity :math:`s_M(x, x')`,
+  defined between two column vectors :math:`x` and :math:`x'` by: :math:
+  `s_M(x, x') =  x M x'`, where :math:`M` is a learned matrix. This matrix
+  is not guaranteed to be symmetric nor positive semi-definite (PSD). Thus
+  it cannot be seen as learning a linear transformation of the original
+  space like Mahalanobis learning algorithms.
+
+  Attributes
+  ----------
+  components_ : `numpy.ndarray`, shape=(n_components, n_features)
+    The learned bilinear matrix ``M``.
+  """
+
+  def score_pairs(self, pairs):
+    dpr_msg = ("score_pairs will be deprecated in release 0.7.0. "
+               "Use pair_score to compute similarity scores, or "
+               "pair_distances to compute distances.")
+    warnings.warn(dpr_msg, category=FutureWarning)
+    return self.pair_score(pairs)
```

Comment:
As this is a new class, I would say that we do not need this method at all as there is no risk that previous code gets broken (since this class did not exist before the switch to `pair_score`).
If you prefer to keep it, then copy the whole docstring, deprecation notice etc as in Mahalanobis case.

Reply:
It's still needed for the tests where all kinds of learners are involved, otherwise an "not implemented method" may appear.

This will change when we remove `socre_pairs` in one or two releses in the future.

I added the docstring